src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -62,32 +71,23 @@ object WorkflowManagerActor {
  * WorkflowOutputs: Returns a `Future[Option[binding.WorkflowOutputs]]` aka `Future[Option[Map[String, WdlValue]]`
  *
  */
-class WorkflowManagerActor(backend: Backend) extends Actor with CromwellActor {
+class WorkflowManagerActor(backend: Backend) extends LoggingFSM[WorkflowManagerState, WorkflowManagerData] with CromwellActor {
   import WorkflowManagerActor._
-  private val log = Logging(context.system, this)
+  private val logger = Logging(context.system, this)
   private val tag = "WorkflowManagerActor"
 
   type WorkflowActorRef = ActorRef
 
   private val workflowStore = new WriteOnceStore[WorkflowId, WorkflowActorRef]
 
-  if (getAbortJobsOnTerminate) {
-    Runtime.getRuntime.addShutdownHook(new Thread() {
-      override def run(): Unit = {
-        log.info(s"$tag: Received shutdown signal. Aborting all running workflows...")
-        workflowStore.toMap.foreach{case (id, actor)=>
-          CromwellServer.workflowManagerActor ! WorkflowManagerActor.WorkflowAbort(id)
-        }
-        var numRemaining = -1
-        while(numRemaining != 0) {
-          Thread.sleep(1000)
-          val result = globalDataAccess.getWorkflowsByState(Seq(WorkflowRunning, WorkflowAborting))
-          numRemaining = Await.result(result,Duration.Inf).size
-          log.info(s"$tag: Waiting for all workflows to abort ($numRemaining remaining).")
-        }
-        log.info(s"$tag: All workflows aborted.")
-      }
-    })
+  private val donePromise = Promise[Unit]()
+
+  if (abortJobsOnTerminate) {
+    sys.addShutdownHook {
```

Comment:
What I meant was why do you need both hooks? The one in the single actor sends WMA a signal to abort its WF and the one here aborts all workflows. If this is just for run (i forget) wouldn't this be redundant and if this is for both run/server wouldn't the other be redundant?